src/wui/constructionsitewindow.h

```diff
@@ -93,17 +93,17 @@ struct ConstructionSiteWindow : public BuildingWindow {
 	};
 
 	Widelands::OPtr<Widelands::ConstructionSite> construction_site_;
-	UI::ProgressBar* progress_;
+	UI::ProgressBar* progress_{nullptr};
 
 	// BuildingSettings-related UI elements
-	UI::Checkbox* cs_launch_expedition_;
+	UI::Checkbox* cs_launch_expedition_{nullptr};
 	std::unique_ptr<UI::Radiogroup> cs_prefer_heroes_rookies_;
```

Comment:
```suggestion
	std::unique_ptr<UI::Radiogroup> cs_prefer_heroes_rookies_{nullptr};
```

Reply:
is deleted in the cc file must be in initialized here

Reply:
No because `unique_ptr` has a default constructor that implicitly initializes as a `nullptr` value. So no initialization of a smart pointer is the same as explicitly setting to nullptr. The recommended way is to omit such initialization, just as `std::string str("")` is discouraged.

Reply:
ok thanks. one reason for doing these big reviews is to learn things like this.